rfcs/testing.md

```diff
@@ -0,0 +1,75 @@
+# RFC: Testing standards
+
+---
+
+_@andrefcdias_
+
+## Summary
+
+The purpose of this RFC is to standardize our testing process and clarify the purpose of each of the testing layers we have. It is supposed to define what is the intended result from tests and why they are required.
+
+## Background
+
+This RFC came from the discussion of unit testing styles in components such as Flex and Text and the debate of having a valuable test vs a "correct" one.
+
+## Problem statement
+
+At the moment, everyone has their own testing standards and there is no common approach to problems like components that only apply styles and need to be tested.
+There is no majority decision on the solution for this problem, being split between Visual Regression tests (VR tests) and Unit Tests (UT).
+
+By defining a common approach, we can make the testing process less ambiguous, as well as provide a guide for the community to follow.
+
+## Detailed Design or Proposal
+
+To start, this is the proposal for defining the goals of each testing layer.
+
+- **Conformance**
+
+  Ensuring the consistency of our components. They test that props like `as`, `ref`, `className` are supported and that the component is rendered.
+
+  These can be achieved simply through the `isConformant` function generated when creating a converged component.
+
+- **Snapshot**
+
+  Stricly DOM structure consistency enforcing. This will guarantee we do not introduce breaking changes for users who expect a certain structure to be rendered by our components.
+
+  They are achieved with Jest's `expect(component).toMatchSnapshot()`.
```

Comment:
can we instead use `toMatchInlineSnapshot()` ? 

while using snapshots is usually an anti-pattern (which also slows down testing execution), when there is a valid use case to use it ( not sure if DOM structure is a valid use-case but that's for different discussion ), inline snaphosts make the test more obvious - you can immediately see what's being asserted etc instead of reaching out to some hidden files (standard snaphosts )

Reply:
Great suggestion, I'll update it! Please remind me, this gets automatically updated with `jest -u` and it's not something we need to manually update, right?

Also, I think it might be relevant to discuss the use case here. What other benefits do we get from snapshot testing if not for avoiding regressions on a DOM level?